modules/apps/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/util/test/CalendarUtilTest.java

```diff
@@ -204,6 +207,49 @@ null, createThemeDisplay(), calendarBookingInstance,
 		assertRepeatsForever(recurrence);
 	}
 
+	@Test
+	public void testToCalendarBookingJSONObjectVulnerabilities()
+		throws Exception {
+
+		ServiceContext serviceContext = createServiceContext();
+
+		HashMap<Locale, String> injectedContent = HashMapBuilder.put(
+			LocaleUtil.getDefault(),
+			"lp'\"></option><img src=x onerror=alert(document.location)>"
+		).build();
+
+		CalendarResource calendarResource =
+			_calendarResourceLocalService.addCalendarResource(
+				TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+				_classNameLocalService.getClassNameId(CalendarResource.class),
+				0, null, null, injectedContent,
+				RandomTestUtil.randomLocaleStringMap(), true, serviceContext);
+
+		Calendar calendar = _calendarLocalService.addCalendar(
+			TestPropsValues.getUserId(), TestPropsValues.getGroupId(),
+			calendarResource.getCalendarResourceId(), injectedContent,
+			RandomTestUtil.randomLocaleStringMap(), StringPool.UTC, 0, false,
+			false, false, serviceContext);
+
+		CalendarBooking calendarBookingInstance =
+			CalendarBookingTestUtil.addRecurringCalendarBooking(
+				TestPropsValues.getUser(), calendar,
+				RecurrenceTestUtil.getDailyRecurrence(), serviceContext);
+
+		Method method = _calendarUtilClass.getMethod(
+			"toCalendarBookingJSONObject", ThemeDisplay.class,
+			CalendarBooking.class, TimeZone.class);
+
+		JSONObject jsonObject = (JSONObject)method.invoke(
+			null, createThemeDisplay(), calendarBookingInstance,
+			calendarBookingInstance.getTimeZone());
+
+		Assert.assertEquals(
+			"lp&#39;&#34;&gt;&lt;/option&gt;&lt;" +
+				"img src=x onerror=alert(document.location)&gt;",
```

Comment:
Hey @rijema apply the same sort above here as well.

Thanks!

Reply:
Done! Same as above.